eclipse-collections/src/main/java/org/eclipse/collections/impl/map/strategy/mutable/UnifiedMapWithHashingStrategy.java

```diff
@@ -361,7 +361,7 @@ protected final int index(K key)
         // This function ensures that hashCodes that differ only by
         // constant multiples at each bit position have a bounded
         // number of collisions (approximately 8 at default load factor).
-        int h = this.hashingStrategy.computeHashCode(key);
+        int h = key == null ? 0 : this.hashingStrategy.computeHashCode(key);
```

Comment:
I thought the null check was purposely skipped and delegated to users to handle the hashCode for null keys. But this predates my involvement in the project, so I could be mistaken.

Reply:
That sounds to be contradicting with UnifiedMap#index(key) implementation. Hopefully @donraab or @motlin could clarify?

```
    protected final int index(Object key)
    {
        // This function ensures that hashCodes that differ only by
        // constant multiples at each bit position have a bounded
        // number of collisions (approximately 8 at default load factor).
        int h = key == null ? 0 : key.hashCode();
        h ^= h >>> 20 ^ h >>> 12;
        h ^= h >>> 7 ^ h >>> 4;
        return (h & (this.table.length >> 1) - 1) << 1;
    }
```

Reply:
it will throw a NPE for the hashCode() call. So, it is necessary to handle that.

Reply:
Ah now I understand what you mean. The decision should be made in each hashingStrategy I guess?

Reply:
precisely